### PR TITLE
New reconcile and UI

### PR DESCRIPTION
**Pull request by adn1107**
Includes the following:
- enhance reconciliation
- new UI
- better detection of duplicate IMSPlex
- reduce runtime
- warm or cold start IMS if normal start fails
- remove dead code
- divide vars file into IMS and z/OS specific file
- capture errors and expose to Conditions messages
- dependencies update

### CODE REVIEW

File: customization-guide.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+It is a good practice to review the playbook contents before executing
+them. It will help you understand the requirements in terms of space, location,
+names, authority, and the artifacts that will be created and cleaned up.
+Although playbooks are written to operate without the need for the user’s
+configuration, flexibility is written into the playbooks because it is not easy
+to determine if a playbook has access to the host’s resources. Review the
+playbook notes sections for additional details and configuration.
+
+## Requirements
+* IBM z/OS core collection 1.5.0
```
Comment (itsBryantP) [inline customization-guide.md:15]: The z/OS Core collection requirement here does not match the `requirements.yml` which states `1.6.0`

File: customization-guide.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+* IBM z/OS IMS collection 1.2.0
+* IBM® Wazi Sandbox 2.4 or IBM® Extended z/OS® ADCD for Z Development and Test Environment built upon the general release of ADCD z/OS® V2R5 December Edition of 2022 or later
```
Comment (itsBryantP) [inline customization-guide.md:17]: Should 2.6 also be mentioned for the October release?

File: galaxy.yml [hunk 3]
```diff
@@ -26,20 +51,30 @@ tags:
   - zos_ims
   - z_and_cloud_modernization_stack_operator
   - zoscb
+
+# Collections that this collection requires to be installed for it to be usable. The key of the dict is the
+# collection label 'namespace.name'. The value is a version range here: https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification. Multiple version
+# range specifiers can be set and are separated by ','
 dependencies:
   ibm.ibm_zos_core: '>=1.3.5'
```
Comment (itsBryantP) [inline galaxy.yml:59]: Does the core collection version need to updated to match the `requirements.yml`?

File: playbooks/delete-ims-datasets.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- name: Delete IMS datasets
+  hosts: all
+  gather_facts: false
+  # * Include different variable files depending on environment/inventory used
+  vars_files:
+    - vars_wazi/ims-dbdc.yml
+    - vars_wazi/ims-zos.yml
+  environment: '{{ system_environment }}'
+  vars:
+    role_path: '/Users/adn/workspaces/nazare/operator-collections/ims/playbooks/roles/ims_initialize'
```
Comment (itsBryantP) [inline playbooks/delete-ims-datasets.yml:15]: `role_path` contains a hardcoded path.

File: playbooks/group_vars/all.yml [hunk 5]
```diff
@@ -2,8 +2,8 @@
 # Copyright 2023 IBM Inc. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 #
-ansible_python_interpreter:  /usr/lpp/IBM/cyp/v3r9/pyz/bin/python3
-PYZ: /usr/lpp/IBM/cyp/v3r9/pyz
+ansible_python_interpreter:  /usr/lpp/IBM/cyp/v3r10/pyz/bin/python3
```
Comment (itsBryantP) [inline playbooks/group_vars/all.yml:5]: Will this 3.10 python path be problematic for wazi sandbox 2.6?

File: playbooks/handlers/handlers.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+---
+# - name: Notify Kubernetes
+#   delegate_to: localhost
```
Comment (itsBryantP) [inline playbooks/handlers/handlers.yml:3]: Is this file needed? Appears to be fully commented out.

File: playbooks/test.yml
Comment (itsBryantP) [inline playbooks/test.yml:1]: Should this test file be included in the final build?

File: inventories/zos_endpoints [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+# Copyright 2023 IBM Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+all:
+  hosts:
+    wazi:
+      ansible_host: your-wazi-hostname
+      ansible_user: ibmuser
+      ansible_port: 2022
+      ansible_python_interpreter:  /usr/lpp/IBM/cyp/v3r10/pyz/bin/python3
```
Comment (itsBryantP) [inline inventories/zos_endpoints:11]: Will this 3.10 python path be problematic for wazi sandbox 2.6?

File: playbooks/filter_plugins/__pycache__/workflows.cpython-311.pyc
Comment (itsBryantP) [inline playbooks/filter_plugins/__pycache__/workflows.cpython-311.pyc:1]: any benefit to including the `pycache` binaries?

File: playbooks/library/submit_jcl.py
Comment (itsBryantP) [inline playbooks/library/submit_jcl.py:1]: this module doesnt appear to be used anywhere. If thats the case it should be deleted.

File: playbooks/library/state_representing_workflow.py
Comment (itsBryantP) [inline playbooks/library/state_representing_workflow.py:1]: this module doesnt appear to be used anywhere. If thats the case it should be deleted.

File: playbooks/library/state_representing_inventory.py
Comment (itsBryantP) [inline playbooks/library/state_representing_inventory.py:1]: this module doesnt appear to be used anywhere. If thats the case it should be deleted.

**Review by itsBryantP — CHANGES_REQUESTED** [review]

**Comment by adn1107** [comment]
@itsBryantP Thanks for the review. The PR was based on an outdated branch before we had to change the dependencies to account for python 3.11 on wazi 2.6. Some other comments still applied so I'll update the latest branch and create a new PR for you to verify.
